pkg/cmd/cli/uninstall/uninstall.go

```diff
@@ -331,22 +412,57 @@ func forcedlyDeleteRestore(ctx context.Context, kbClient kbclient.Client, namesp
 		return errors.Wrap(err, "Error deleting velero deployment during force deletion")
 	}
 
-	// Remove all the restores' finalizer so they can be deleted during the deletion of velero namespace.
-	restoreList := &velerov1api.RestoreList{}
-	if err := kbClient.List(ctx, restoreList, &kbclient.ListOptions{Namespace: namespace}); err != nil {
-		return errors.Wrap(err, "Error getting restores during force deletion")
-	}
+	return removeResourceFinalizer(ctx, kbClient, namespace)
+}
+
+func removeResourceFinalizer(ctx context.Context, kbClient kbclient.Client, namespace string) error {
```

Comment:
For this function, I try to use `reflect` in Golang, It always panics with `panic: interface conversion: v2alpha1.DataUpload is not client.Object: missing method DeepCopyObject`. I could not figure it out in a short time. so I keep it with redundancy codes

```golang
func removeFinalizer(ctx context.Context, kbClient kbclient.Client, namespace string, resourceList kbclient.ObjectList, finalizer string) error {
	listOptions := &kbclient.ListOptions{Namespace: namespace}
	if err := kbClient.List(ctx, resourceList, listOptions); err != nil {
		return errors.Wrap(err, fmt.Sprintf("Error getting resources of type %T during force deletion", resourceList))
	}

	itemsValue := reflect.ValueOf(resourceList).Elem().FieldByName("Items")
	for i := 0; i < itemsValue.Len(); i++ {
		item := itemsValue.Index(i).Interface().(kbclient.Object)
		if controllerutil.ContainsFinalizer(item, finalizer) {
			update := item.DeepCopyObject().(kbclient.Object)
			original := item.DeepCopyObject().(kbclient.Object)
			controllerutil.RemoveFinalizer(update, finalizer)
			if err := kubeutil.PatchResource(original, update, kbClient); err != nil {
				return errors.Wrap(err, fmt.Sprintf("Error removing finalizer %q during force deletion", finalizer))
			}
		}
	}
	return nil
}
```

Reply:
If our type does not conform to that interface, I think this could be a larger problem for anyone working with these API's. I think to fix, you have to run the controller-tools tool to generate these methods for the API.

https://book.kubebuilder.io/reference/markers/object.html

Reply:
Thanks for your comment @shawn-hurley, I've fixed it.
I should not covert interface into `kbclient.Object`type  in `item := itemsValue.Index(i).Interface().(kbclient.Object)`

As here is the definition of DatauploadList 
```golang
type DataUploadList struct {
    metav1.TypeMeta `json:",inline"`

    // +optional
    metav1.ListMeta `json:"metadata,omitempty"`

    Items []DataUpload `json:"items"`
}
```
And we have the DeepCopyObject function for *DataUpload type not DataUpload:
```golang
func (in *DataUpload) DeepCopyObject() runtime.Object {
	if c := in.DeepCopy(); c != nil {
		return c
	}
	return nil
}
```
The Item in DataUploadList is `DataUpload` but not `*DataUpload`, so it panic with `v2alpha1.DataUpload is not client.Object: missing method DeepCopyObject`